[Keycloak-Operator]: Rolling Updates -- Strategy=Auto, operator error keycloak-update-job is invalid -- Strategy=Explicit, operator always replaces

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

operator

### Describe the bug

On the latest `v26.2.5` of `keycloak-k8s-resources`, `operator` and `RedHat Build of Keycloak Server`. Keycloak-Operator rolling updates are not working as expected. When `spec.update.strategy=auto`, the operator fails with error: `keycloak-update-job is invalid`, and when `spec.update.strategy=Explicit`, the operator will always replace the pod; rolling update is not observed. 

### Version

26.2.5

### Regression

- [ ] The issue is a regression

### Expected behavior

1. When Keycloak's CRD has `spec.update.strategy=auto`, the operator should automatically determine the best strategy to deploy a new update of the Keycloak server, and default to `RecreateOnImageChange` if the checks fail.
2. When Keycloak's CRD has `spec.update.strategy=Explicit`, the operator should deploy a new update of the Keycloak server by rolling deployment, when `spec.update.revision` matches the previous update. 

### Actual behavior

### When Keycloak's CRD has `spec.update.strategy=auto`:
Operator throw the following error: 
```shell
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: PATCH at: https://172.20.0.1:443/apis/batch/v1/namespaces/identity/jobs/keycloak-update-job?fieldManager=keycloakcontroller&force=true. Message: Job.batch "keycloak-update-job" is invalid: │
│  spec.template.spec.initContainers[0].lifecycle: Forbidden: may not be set for init containers without restartPolicy=Always. Received status: Status(apiVersion=v1, code=422, details=StatusDetails(causes=[StatusCause(field=spec.template.spec.initContainers[0].lifecycle, mess │
│ age=Forbidden: may not be set for init containers without restartPolicy=Always, reason=FieldValueForbidden, additionalProperties={})], group=batch, kind=Job, name=keycloak-update-job, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=Job.batch │
│  "keycloak-update-job" is invalid: spec.template.spec.initContainers[0].lifecycle: Forbidden: may not be set for init containers without restartPolicy=Always, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalPropertie │
│ s={}), reason=Invalid, status=Failure, additionalProperties={}).
```

When the error is thrown, the operator will not deploy a new version of the Keycloak Server Pod until  `spec.update` is removed or the `strategy` is changed. 


### When Keycloak's CRD has `spec.update.strategy=Explicit`
The operator will always deploy a new Keycloak Server Pod by replacing it - no rolling update is ever observed. 

### How to Reproduce?

Deploy keycloak-k8s-resources: `v26.2.5` as usual and start updating the Keycloak Server image property with new tags.  

### Anything else?

System Details:
keyclaok-operator version: `v26.2.5`
RedHat Build of Keycloak: `v26.2.5`
keycloak-k8s-resources: `v26.2.5`
AWS EKS Automode - Kubernetes: `v1.32`

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakUpdateJobDependentResource.java
@@ -193,8 +193,9 @@ public class KeycloakUpdateJobDependentResource extends CRUDKubernetesDependentR
             containerBuilder.withVolumeMounts(newVolumeMounts);
         }
 
-        // remove restart policy and probes
+        // remove restart policy, lifecycle, and probes
         containerBuilder.withRestartPolicy(null);
+        containerBuilder.withLifecycle(null);
         containerBuilder.withReadinessProbe(null);
         containerBuilder.withLivenessProbe(null);
         containerBuilder.withStartupProbe(null);

--- a/operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
+++ b/operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java
@@ -70,6 +70,7 @@ import java.util.stream.Collectors;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.keycloak.operator.ContextUtils.DIST_CONFIGURATOR_KEY;
 import static org.keycloak.operator.ContextUtils.NEW_DEPLOYMENT_KEY;
@@ -706,6 +707,18 @@ public class PodTemplateTest {
 
         job = getUpdateJob(builder -> builder.addToImagePullSecrets(new LocalObjectReference("new-secret")), builder -> {}, addSecret);
         assertEquals(List.of(new LocalObjectReference("new-secret"), new LocalObjectReference("secret")), job.getSpec().getTemplate().getSpec().getImagePullSecrets());
+    }
+
+    @Test
+    public void testFieldRemovalForInitContainer() {
+        Job job = getUpdateJob(builder -> {
+        }, builder -> builder.withNewUnsupported().withNewPodTemplate().withNewSpec().addNewContainer()
+                .withRestartPolicy("OnFailure")
+                .withNewLifecycle().withNewPostStart().withNewExec().withCommand("hello").endExec().endPostStart()
+                .endLifecycle().endContainer().endSpec().endPodTemplate().endUnsupported(), builder -> {
+                });
+        assertNull(job.getSpec().getTemplate().getSpec().getInitContainers().get(0).getLifecycle());
+        assertNull(job.getSpec().getTemplate().getSpec().getInitContainers().get(0).getRestartPolicy());
     }
 
 }

